best_practices.md

```diff
@@ -0,0 +1,233 @@
+[style guide](https://the-turing-way.netlify.app/reproducible-research/code-quality.html)
+that is available for the programming language of your choice.
+Examples of style guides:
+- [PEP8](https://www.python.org/dev/peps/pep-0008/) and [PEP257](https://www.python.org/dev/peps/pep-0257/) for Python
+- The [Tidyverse style guide ](https://style.tidyverse.org/)for R
+
+If there is no style guide available for the programming language of your
+choice, you may want to reconsider that choice because the language may not be
+mature enough.
+If there are compelling reasons to choose a language without a style guide,
+it is recommended that you at least
+- organize your code in multiple files of no more than a few hundred lines each
+- organize your code in functions that fit your screen without scrolling (typically no more than 50 lines)
+- use easy to understand variable names
+- insert documentation or comments in your code that explain what is going on
+
+This kind of organization increases readability of your code, thereby reducing
+the risk of mistakes and making your code accessible to others.
+
+Style guides typically come with automatic formatters, that can automatically
+format your code according to the chosen style.
+In addition to automatic formatters, tools for checking adherence to a style
+guide as well as for common programming mistakes are usually also available.
+These tools are called linters.
+It is recommended that you make use of an automatic formatter and linter.
+For example, a popular automatic formatter for Python is
+[black](https://black.readthedocs.io), while a good linter for Python is
+[pylint](https://pylint.org/).
+It is recommended that you use an [online service to host your code](#host-your-code-on-a-website-with-an-issue-tracker),
+so you can [automate checking the quality of your code](#use-a-code-quality-checker)
+and even [formatting it](https://pre-commit.ci), should you wish to do so.
+
+## Keep your code reliable and maintainable
+If you write large pieces of code, it is recommended that you implement some
+form of
+[automated testing](https://the-turing-way.netlify.app/reproducible-research/testing.html)
+to ensure that you do not loose more and more time testing that your code still
+produces the correct results as it grows.
+This can range from
+[unit tests](https://the-turing-way.netlify.app/reproducible-research/testing/testing-unittest.html),
+that test each function in your code individually, to
+[regression tests](https://the-turing-way.netlify.app/reproducible-research/testing/testing-acceptance-regression.html#regression-testing)
+that run larger portions of code and just check that the changes you make to your
+code do not change previously computed results that are not supposed to change.
+For most programming languages, tools are available to run all tests with a
+single command.
+A good choice for running Python tests is [pytest](https://docs.pytest.org).
```

Comment:
I feel that a comment about tools to check the coverage of the tests its in order here. It may seem obvious for us, but I bet most scientists will not even think about it

Reply:
Thanks, added